### PR TITLE
tests: ringbuffer, verify the array stored by ringbuffer

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add new informative Doxygen tags for the ring buffer tests.
1.test_ring_buffer_main()
Test APIs ring_buf_item_put(),ring_buf_item_get(), and verify that ring buffer put data will return error when full, identically, get data will return error when empty.Add Doxygen tag describing that.

2.test_ringbuffer_raw()
Test that data is passed between ring buffer and array by APIs ring_buf_put(),ring_buf_get(). Add Doxygen tag describing that.

3.add a new testcase:test_ringbuffer_array_perf()
Test that the address of array stored by ringbuffer is contiguous, meanwhile, the size of the array is equal to the size of a pointer, as more as possible to compatible about all platforms(i.e. 32bits,64bits...).

4.Modify some code style.
Add blank lines or reduce indentation to ensure less than 80 bytes per line.

Signed-off-by: Ningx Zhao ningx.zhao@intel.com

### CODE REVIEW

File: tests/lib/ringbuffer/src/main.c [hunk 1]
```diff
@@ -443,25 +476,76 @@ void test_reset(void)
...
-			 ztest_unit_test(test_ringbuffer_raw),
-			 ztest_unit_test(test_ringbuffer_alloc_put),
-			 ztest_unit_test(test_byte_put_free),
-			 ztest_unit_test(test_byte_put_free),
-			 ztest_unit_test(test_capacity),
-			 ztest_unit_test(test_reset)
-			 );
+		 ztest_unit_test(test_ringbuffer_init),/*keep init first!*/
+		 ztest_unit_test(test_ringbuffer_declare_pow2),
+		 ztest_unit_test(test_ringbuffer_declare_size),
```
Comment (andyross) [inline tests/lib/ringbuffer/src/main.c:535]: Please make your new changes conform to the whitespace conventions of the original so we can tell what changed.  If you want to reflow a big region like this, make that change a separate patch.

File: tests/lib/ringbuffer/src/main.c [hunk 2]
```diff
@@ -443,25 +476,76 @@ void test_reset(void)
...
+
+	zassert_true(ring_buf_item_put(&buf_ii, 1, 2, input, 3) == 0, NULL);
+
+	/*verify the address stored by ringbuf is contiguous*/
+	for (int i = 0; i < 3; i++) {
+		zassert_equal(input[i], buf_ii.buf.buf32[i+1], NULL);
+	}
+
+	/*Offset to the end of the in-use data in the array*/
+	while (output[count] != 0) {
```
Comment (andyross) [inline tests/lib/ringbuffer/src/main.c:513]: Uh... why are we doing this?

You're initializing an output buffer on the stack with some non-zero data at the start, skipping over the non-zero bytes (which you just put there!), and then using the suffix of the buffer as the argument to ring_buf_item_get().  Why?  What's the condition being tested here?  The ring buffer doesn't care what's in the memory before the buffer you passed.

I mean, maybe you're intending to test that it won't write before the buffer you passed?  But if so, you don't actually check that condition; the prefix bytes are ignored.

**Review by andyross — CHANGES_REQUESTED** [review]
A whitespace whine that managed to get past my threshold for a -1.  Also something wonky is going on with those extra array elements and needs some elaboration or removal.

File: tests/lib/ringbuffer/src/main.c (hunk 1 above)
Comment (Zhaoningx) [reply tests/lib/ringbuffer/src/main.c]: Hi @andyross , thanks for your comments, you are right, I shouldn't change this, so I have regained this region.

But there are something confused me, I used my native tools to check this region and it showed me "line over 80 characters", so I modified it. Actually,I find that it can be pushed to github successfully without modifying this.

File: tests/lib/ringbuffer/src/main.c (hunk 2 above)
Comment (Zhaoningx) [reply tests/lib/ringbuffer/src/main.c]: Hi @andyross , Uh... maybe I'm misleading by my thinking. Actually, this operation skipping non-zero data of output buffer maybe is clumsy and confusing. And this test point is covered by the other region of this testcase, so I have removed this code you mentioned. Thanks for your comments again.

**Review by andyross — APPROVED** [review]

File: tests/lib/ringbuffer/src/main.c [hunk 3]
```diff
@@ -443,6 +476,58 @@ void test_reset(void)
 	zassert_true(granted == RINGBUFFER_SIZE - 1, NULL);
 }
 
+#ifdef CONFIG_64BIT
+static uint64_t ringbuf_stored[RINGBUFFER_SIZE];
```
Comment (mbolivar-nordic) [inline tests/lib/ringbuffer/src/main.c:480]: Is this correct? The ring_buf_init doxygen says:

```
 * @param data Ring buffer data area (uint32_t data[size] or uint8_t data[size] for
 *	       bytes mode).
```

Comment (andyross) [reply tests/lib/ringbuffer/src/main.c:480]: Good catch.  Yeah, this is wasteful (though not incorrect as I read the code).  If you're going to use extra memory in a test like this, at least do it to put a guard area at the edges to check for overrruns. :)

**Review by mbolivar-nordic — APPROVED** [review]
Seems sane (I'm trying to help out with the new 2 reviewers requirement today), will leave the guard question up to @andyross

File: tests/lib/ringbuffer/src/main.c (hunk 3 above)
Comment (Zhaoningx) [reply tests/lib/ringbuffer/src/main.c:480]: Hi @mbolivar-nordic @andyross , thanks for your comments, yeah, I totally agree that it's not allowed to use the array defined by uint64_t in normal. Notably, this is a test that the pointer size is equal to the ring buffer size. And the platforms are 32bits and 64bits, the pointer also. To keep with the same as pointer size, I use the #ifdef#endif to ensure that testcase can be implement. 

By the way, the ring buffer is implemented as a 'ring', so why do I need to put a guard area? Thanks again. :)

Comment (Zhaoningx) [reply tests/lib/ringbuffer/src/main.c:480]: Hi @mbolivar-nordic @andyross , thanks for your comments, yeah, I totally agree that it's not allowed to use the array defined by uint64_t in normal. Notably, this is a test that the pointer size is equal to the ring buffer size. And there are 32bits and 64bits about platforms(and pointer). To keep with the same as pointer size, I used the #ifdef#endif to ensure that the testcase is compatible with all platforms. 

And... It's actually wasteful, I find that the extra memory can't be used by ring buffer APIs in 64bits platform. And the test is just to testify that ring buffer with (32bits or 64bits)buffer can work. :)

Comment (andyross) [reply tests/lib/ringbuffer/src/main.c:480]: You don't need a guard area.  I was saying that **if** you were going to waste memory like this with an oversized buffer, one **good** reason (in a test like this) would be to use it to test for overruns by the code.

Comment (Zhaoningx) [reply tests/lib/ringbuffer/src/main.c:480]: Got it, thanks very much. :)

**Review by ioannisg — APPROVED** [review]
